u-boot/cpu/mips/ar7240/qca_dram.c

```diff
@@ -781,16 +802,18 @@ static inline void qca_dram_set_en_refresh(void)
 	if (qca_xtal_is_40mhz()) {
 		qca_soc_reg_write(QCA_DDR_REFRESH_REG,
 				  QCA_DDR_REFRESH_EN_MASK
-				  | (312 << QCA_DDR_REFRESH_PERIOD_SHIFT));
+				  | (312UL << QCA_DDR_REFRESH_PERIOD_SHIFT));
 	} else {
 		qca_soc_reg_write(QCA_DDR_REFRESH_REG,
 				  QCA_DDR_REFRESH_EN_MASK
-				  | (195 << QCA_DDR_REFRESH_PERIOD_SHIFT));
+				  | (195UL << QCA_DDR_REFRESH_PERIOD_SHIFT));
 	}
 }
 
 /*
  * Initial DRAM configuration
+ *
+ * Not working on QCA956x
```

Comment:
@j-d-r What does not work?
In TP-Link's source code the CAS latency is being set manually for AP151 (CFG_DDR2_DRAGONFLY_CAS_LATENCY). Some other things also seem to differ, but I don't want to dive deep into it if you can explain your findings.

Reply:
Hello, I've forgotten this comment when I merged code from tp-link u-boot into pepe2k dram code. https://github.com/j-d-r/u-boot-QCA956x/commit/364d623526ee2bc0f8919d0f4524fc3a8ce91d00